### PR TITLE
Documentation improvements to the times module

### PR DESCRIPTION
**Pull request by ssorc**
Updates to the documentation for the times module: improved grammar, corrected spelling and typos, added some clarifications.

### CODE REVIEW

File: lib/pure/times.nim [hunk 1]
```diff
@@ -895,13 +895,13 @@ proc `-`*(a: Time, b: Duration): Time {.operator, extern: "ntSubTime".} =
   subImpl[Time](a, b)
 
 proc `<`*(a, b: Time): bool {.operator, extern: "ntLtTime".} =
-  ## Returns true iff ``a < b``, that is iff a happened before b.
+  ## Returns true if ``a < b``, that is if ``a`` happened before ``b``.
```
Comment (GULPF) [inline lib/pure/times.nim]: `iff` is not a typo, see https://en.wikipedia.org/wiki/If_and_only_if

Comment (narimiran) [reply lib/pure/times.nim]: > `iff` is not a typo

...but this keeps popping up very frequently, so I'm not against changing it so at least in the future there won't be more of these "fixes" anymore.

Comment (ssorc) [reply lib/pure/times.nim]: I appreciate that "iff" is not a typo, but it is not common English and it is not clear why it is explicitly required for these functions' documentation over making the documentation more readable and accessible, particularly for readers who do not have English as their native language. If there is a need to definitely use "iff" here (which possibly suggests that we should also indicate why that is the case), then I can back out those changes to ensure the rest of the PR is accepted. Thanks!

Comment (zestyr) [reply lib/pure/times.nim]: Note that "iff" is used 36 times throughout the stdlib's documentation

**Comment by Araq** [comment]
Thank you!

**Pull request merged**